services/f-droid/f-droid.tester.js

```diff
@@ -8,169 +8,74 @@ const t = (module.exports = new ServiceTester({
-  .expectBadge({ label: 'f-droid', message: 'v1.4' })
-
-t.create('Trailing 0 in yml format')
-  .get('/v/axp.tool.apkextractor.json')
-  .intercept(nock => nock(base).get(`${path}.txt`).reply(404))
-  .intercept(nock =>
-    nock(base).get(`${path}.yml`).reply(200, "CurrentVersion: '1.4000'")
-  )
-  .expectBadge({ label: 'f-droid', message: 'v1.4000' })
-
-t.create('Package is found with yml matadata format')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=yml')
-  .intercept(nock => nock(base).get(`${path}.yml`).reply(200, testYmlString))
-  .expectBadge({ label: 'f-droid', message: 'v1.4' })
+t.create('Package is not found with 403')
+  .get(`/v/${testPkg}.json`)
+  .intercept(nock => nock(base).get(path).reply(403, 'some 403 text'))
+  .expectBadge({ label: 'f-droid', message: 'app not found' })
 
-t.create('Package is not found with "metadata_format" query parameter')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=yml')
-  .intercept(nock => nock(base).get(`${path}.yml`).reply(404))
+t.create('Package is not found with 404')
+  .get('/v/io.shiels.does.not.exist.json')
   .expectBadge({ label: 'f-droid', message: 'app not found' })
 
-t.create('Package is found yml matadata format with missing "CurrentVersion"')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=yml')
+t.create('Package is not found with no packages available (empty array)"')
+  .get(`/v/${testPkg}.json`)
   .intercept(nock =>
-    nock(base).get(`${path}.yml`).reply(200, 'Categories: System')
+    nock(base)
+      .get(path)
+      .reply(200, `{"packageName":"${testPkg}","packages":[]}`)
   )
-  .expectBadge({ label: 'f-droid', message: 'invalid response data' })
+  .expectBadge({ label: 'f-droid', message: 'no packages found' })
 
-t.create('Package is found with bad yml matadata format')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=yml')
+t.create('Package is not found with no packages available (missing array)"')
+  .get(`/v/${testPkg}.json`)
   .intercept(nock =>
-    nock(base).get(`${path}.yml`).reply(200, '.CurrentVersion: 1.4')
+    nock(base).get(path).reply(200, `{"packageName":"${testPkg}"}`)
   )
-  .expectBadge({ label: 'f-droid', message: 'invalid response data' })
+  .expectBadge({ label: 'f-droid', message: 'no packages found' })
 
-t.create('Package is not found')
-  .get('/v/axp.tool.apkextractor.json')
-  .intercept(nock => nock(base).get(`${path}.txt`).reply(404))
-  .intercept(nock => nock(base).get(`${path}.yml`).reply(404))
-  .expectBadge({ label: 'f-droid', message: 'app not found' })
-
-t.create('The api changed')
-  .get('/v/axp.tool.apkextractor.json?metadata_format=yml')
-  .intercept(nock => nock(base).get(`${path}.yml`).reply(200, ''))
+t.create('Package is not found with invalid response"')
+  .get(`/v/${testPkg}.json`)
+  .intercept(nock => nock(base).get(path).reply(200, '{"field":"value"}'))
   .expectBadge({ label: 'f-droid', message: 'invalid response data' })
```

Comment:
We don't typically have mocked service tests that inject a schema-failing response. I suppose I don't feel too strongly about it, but just sharing for awareness in case you ever find yourself working in the Shields codebase again :smile:

Reply:
So, I can just remove it, right?